lightsim2grid/newtonpf/newtonpf.py

```diff
@@ -306,14 +307,16 @@ def newtonpf_new(Ybus, Sbus, V0, ref, pv, pq, ppci, options):
 
     # initialize the solver
     # TODO have that in options maybe (can use GaussSeidel, and NR with KLU -faster- or SparseLU)
-    if KLU_solver_available:
-        solver = KLUSolver()
+    if options["distributed_slack"]:
+        solver = KLUSolver() if KLU_solver_available else SparseLUSolver()
     else:
-        solver = SparseLUSolver()
-    Ybus = sparse.csc_matrix(Ybus)
+        solver = KLUSolverSingleSlack() if KLU_solver_available else SparseLUSolverSingleSlack()
+
+    if ~sparse.isspmatrix_csc(Ybus):
+        Ybus = sparse.csc_matrix(Ybus)
```

Comment:
We had also a problem with the "canonical form" if I remember correctly. Is it checked here ?

Reply:
I will add the check

Reply:
Do you think this is adequate, or do you expect that Ybus can also come from a different software?

Reply:
I will add some warnings in the documentation and a check here that will raise an error if it's not in canonical form so it's fine like this